crates/c-api/CMakeLists.txt

```diff
@@ -1,8 +1,9 @@
 cmake_minimum_required(VERSION 3.12)
-project(wasmtime C)
+project(wasmtime)
 
 set(WASMTIME_USER_CARGO_BUILD_OPTIONS "" CACHE STRING "Additional cargo flags (such as --features) to apply to the build command")
 option(BUILD_SHARED_LIBS "Build using shared libraries" OFF)
+option(BUILD_TESTS "Build tests" ON)
```

Comment:
Could this default to OFF but we be careful to pass it as ON in CI? Agreed though that this is good to have, thanks for adding this!